src/simulator_core/adapter/transforms/esdl_asset_mapper.py

```diff
@@ -17,13 +19,20 @@ class EsdlAssetMapper(EsdlMapperAbstract):
     }
 
     def to_esdl(self, entity: AssetAbstract) -> Any:
-        """ Maps entity object to PyEsdl objects"""
+        """Maps entity object to PyEsdl objects."""
         raise NotImplementedError("EsdlAssetMapper.to_esdl()")
 
-    def to_entity(self, model: EsdlAssetObject) -> AssetAbstract:
+    def to_entity(self, model: EsdlAssetObject, **kwargs) -> AssetAbstract:
```

Comment:
We should refactor this.   The required info (from_junction, to_junction, etc)  should already be available within the EsdlAssetObject

Reply:
As discussed this has been refactored